### PR TITLE
Wrap errors for NodeLabel, NodePorts, NodePreferAvoidPods and NodeResourcesBalancedAllocation plugins

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Wrap errors for NodeLabel, NodePorts, NodePreferAvoidPods and NodeResourcesBalancedAllocation plugins

**Which issue(s) this PR fixes**:

Ref  #94695

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by SataQiu** [comment]
Can you take a look? Any suggestions? @alculquicondor

File: pkg/scheduler/framework/plugins/nodelabel/node_label.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func (pl *NodeLabel) Filter(ctx context.Context, _ *framework.CycleState, pod *v
 func (pl *NodeLabel) Score(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
 	nodeInfo, err := pl.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
 	if err != nil || nodeInfo.Node() == nil {
-		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("getting node %q from Snapshot: %v, node is nil: %v", nodeName, err, nodeInfo.Node() == nil))
+		return 0, framework.AsStatus(fmt.Errorf("getting node %q from Snapshot: %w, node is nil: %v", nodeName, err, nodeInfo.Node() == nil))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/nodelabel/node_label.go:113]: We don't need the check `nodeInfo.Node() == nil` anymore. Also, `%w` should always be at the end of the string.

File: pkg/scheduler/framework/plugins/nodeports/node_ports.go [hunk 2]
```diff
@@ -87,7 +87,7 @@ func getPreFilterState(cycleState *framework.CycleState) (preFilterState, error)
 	c, err := cycleState.Read(preFilterStateKey)
 	if err != nil {
 		// preFilterState doesn't exist, likely PreFilter wasn't invoked.
-		return nil, fmt.Errorf("error reading %q from cycleState: %v", preFilterStateKey, err)
+		return nil, fmt.Errorf("error reading %q from cycleState: %w", preFilterStateKey, err)
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/nodeports/node_ports.go:90]: remove the word error. That is implicit from the type

File: pkg/scheduler/framework/plugins/nodeports/node_ports_test.go [hunk 3]
```diff
@@ -170,7 +171,7 @@ func TestPreFilterDisabled(t *testing.T) {
 	p, _ := New(nil, nil)
 	cycleState := framework.NewCycleState()
 	gotStatus := p.(framework.FilterPlugin).Filter(context.Background(), cycleState, pod, nodeInfo)
-	wantStatus := framework.NewStatus(framework.Error, `error reading "PreFilterNodePorts" from cycleState: not found`)
+	wantStatus := framework.AsStatus(fmt.Errorf(`error reading "PreFilterNodePorts" from cycleState: %w`, fmt.Errorf("not found")))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/nodeports/node_ports_test.go:174]: remove word error

Comment (SataQiu) [reply pkg/scheduler/framework/plugins/nodeports/node_ports_test.go]: Thank you very much for your review @alculquicondor
All the Suggestions have been applied.

**Comment by alculquicondor** [comment]
/approve
/lgtm

Thanks!

**Pull request merged**